tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Collections/ArrayModule2.fs

```diff
@@ -644,7 +644,57 @@ type ArrayModule2() =
         Assert.AreEqual([|8;8;8|], eights)
         
         ()   
+
+
+    member private _.MultiplyArray(template:_[],repetitions:int) = 
+        Array.zeroCreate repetitions |> Array.collect (fun _ -> template)     
+
+    member private _.CompareTwoMethods<'TIn,'TOut when 'TOut: equality> (regularArrayFunc:'TIn[]->'TOut[]) (arrayParaFunc:'TIn[]->'TOut[]) (initialData:'TIn[]) = 
+        let first,second = initialData, Array.copy initialData
+        let whenSequential = regularArrayFunc second
+        let whenParallel = arrayParaFunc first
+
+        if(whenSequential <> whenParallel) then
+            Assert.AreEqual(whenSequential.Length, whenParallel.Length, "Lengths are different")
+            let diffsAt = 
+                Array.zip whenSequential whenParallel
+                |> Array.mapi (fun idx (a,b) -> if(a <> b) then Some(idx,(a,b)) else None)
+                |> Array.choose id
+                |> dict
+            
+            Assert.Empty(diffsAt)
+            Assert.Equal<'TOut>(whenSequential, whenParallel)
+
+    [<Fact>]
+    member this.sortInPlaceWithParallel() =      
+
+        let tee f x = f x; x
+        // integer array  
+        this.MultiplyArray([|3;5;7;2;4;8|],1_000) 
+        |> this.CompareTwoMethods (tee (Array.sortInPlaceWith compare)) (tee (Array.Parallel.sortInPlaceWith compare))
+
+        // Sort backwards
+        this.MultiplyArray([|3;5;7;2;4;8|],1_000) 
+        |> this.CompareTwoMethods (tee (Array.sortInPlaceWith (fun a b -> -1 * compare a b))) (tee (Array.Parallel.sortInPlaceWith (fun a b -> -1 * compare a b)))
         
+        // string array
+        let strArr = [|"Lists"; "are"; "a"; "commonly"; "used"; "data"; "structure"|]    
+        this.MultiplyArray(strArr,1_000) 
+        |> this.CompareTwoMethods (tee (Array.sortInPlaceWith compare)) (tee (Array.Parallel.sortInPlaceWith compare))
+        
+        // empty array
+        [| |]
+        |> this.CompareTwoMethods (tee (Array.sortInPlaceWith compare)) (tee (Array.Parallel.sortInPlaceWith compare))       
+        
+        // null array
+        let nullArr = null:string[]      
+        CheckThrowsArgumentNullException (fun () -> Array.Parallel.sortInPlaceWith compare nullArr  |> ignore)  
+
+        // Equal elements              
+        this.MultiplyArray([|8; 8;8|],1_000) 
+        |> this.CompareTwoMethods (tee (Array.sortInPlaceWith compare)) (tee (Array.Parallel.sortInPlaceWith compare))
+        
+        ()           
```

Comment:
Why do all these tests have the extra `()` at the end?

Reply:
Might be historic - perharps some test runners/frameworks only worked with void test methods?
But I assume both nunit and xunit do not care about that (at least anymore), right?

Reply:
I would understand that, but it looks like they would return `unit` anyway even without the extra `()` so I wonder if there could be any other reason... anyway, not important.